client/src/app.ts

```diff
@@ -0,0 +1,121 @@
+import {UserAction , Role} from "./models/model.js";
+
+function userCreatedDate(){
+    return function(target: any,key:string,descriptor:PropertyDescriptor){
+        let val = descriptor.value
+        descriptor.value = function(... args:any[]){
+            let arr:Array<string> = [];
+            for(let j=0;j<7;j++){
+                arr.push(args[0].target[j].value)
+            }
+            let d: Date = new Date();
+            let dt = d.getDate()+" "+d.toLocaleString('en-US', {month: 'long',})+" "+d.getFullYear()+" Time: "+d.getHours()+":"+d.getMinutes();
+            arr.push(dt.toString())
+            return val.apply(this,[arr]);
+        }
+    }
+}
+
+export class User<T extends Array<object>,U extends Array<Array<string>>> implements UserAction{
+    APIData:Array<object> = [];
```

Comment:
user camel case for variables

Reply:
all variables are changed into camel case